tests/unit/session_test.py

```diff
@@ -18,4 +24,55 @@ def test_session_connection_refused(free_port_num):
     error_msg = '^ssh connect failed: Connection refused$'
     ssh_session = Session()
     with pytest.raises(LibsshSessionException, match=error_msg):
-        ssh_session.connect(host='127.0.0.1', port=free_port_num)
+        ssh_session.connect(host=LOCALHOST, port=free_port_num)
+
+
+def test_session_log_level_debug(caplog, free_port_num):
+    """Test setting the log level to DEBUG should reveal copyright information."""
+    ssh_session = Session()
+    ssh_session.set_log_level(logging.DEBUG)
+
+    # the connection will fail but first log lands before that
+    with pytest.raises(LibsshSessionException, match='ssh connect failed: Connection refused'):
+        ssh_session.connect(host=LOCALHOST, port=free_port_num)
+
+    expected_copyright_substring = 'and libssh contributors.'
+    # This log message is shown at different log levels
+    # in different libssh versions:
+    expected_copyright_log_levels = {'DEBUG', 'INFO'}
+    informational_log_messages = (
+        record.msg
+        for record in caplog.records
+        if record.levelname in expected_copyright_log_levels
+    )
+    assert any([expected_copyright_substring in message for message in informational_log_messages])
+
+
+def test_session_log_level_no_log(caplog, free_port_num):
+    """Test setting the log level to CRITICAL should be quiet (unless there is something critical)."""
+    ssh_session = Session()
+    ssh_session.set_log_level(logging.CRITICAL)
+
+    # the connection will fail but first log lands before that
+    with pytest.raises(LibsshSessionException, match='ssh connect failed: Connection refused'):
+        ssh_session.connect(host=LOCALHOST, port=free_port_num)
+
+    assert not caplog.records
+
+
+def test_session_log_level_trace(caplog, free_port_num):
+    """Test setting the log level to TRACE should provide all of the logs."""
+    ssh_session = Session()
+    ssh_session.set_log_level(ANSIBLE_PYLIBSSH_TRACE)
+
+    # the connection will fail but first log lands before that
+    with pytest.raises(LibsshSessionException, match='ssh connect failed: Connection refused'):
+        ssh_session.connect(host=LOCALHOST, port=free_port_num)
+
+    expected_poll_message = 'ssh_socket_pollcallback: Poll callback on socket'
+    informational_log_messages = (
+        record.msg
+        for record in caplog.records
+        if record.levelname == 'TRACE'
+    )
+    assert any([expected_poll_message in message for message in informational_log_messages])
```

Comment:
The intermediate list is unnecessary here. `any()` works with generator expressions just fine.
```suggestion
    assert any(expected_poll_message in message for message in informational_log_messages)
```

Reply:
Fixed! Thanks!